src/container/FinishSignUp/useFinishSignUp.ts

```diff
@@ -0,0 +1,154 @@
+  const [allFieldsData, setAllFieldsData] = useState<IField[]>([])
+  const [frameWorks, setFrameWorks] = useState<IFrameWork[]>([])
+  const { loading, getHandler, postHandler } = useRequestHandlers()
+  const t = useTranslations('finishSignup')
+  const {
+    formState: { errors },
+    control,
+    handleSubmit,
+    watch,
+  } = useForm<IFinishSignUpFrom>({ resolver: yupResolver(schema), defaultValues })
+  const tracksValue = watch('tracks')
+  const list: IQuestionsList[] = [
+    {
+      Text: 'What is your current employment status?',
+      name: 'tracks',
+      QuestionChoices: allFieldsData,
+    },
+    {
+      Text: 'Which technology/framework have you learned? *',
+      name: 'frameworks',
+      QuestionChoices: frameWorks,
+    },
+    {
+      Text: 'What is your current employment status?',
+      name: 'employmentStatus',
+      QuestionChoices: [
+        { name: 'full time', id: 'full time' },
+        { name: 'part time', id: 'part time' },
+        { name: 'Student', id: 'Student' },
+        { name: 'unemployed', id: 'unemployed' },
+      ],
+    },
+    {
+      Text: 'In which language you prefer to learn?',
+      name: 'preferredLanguage',
+      QuestionChoices: [
+        { name: 'arabic', id: 'arabic' },
+        { name: 'english', id: 'english' },
+      ],
+    },
+    {
+      Text: 'How did you hear about Topera?',
+      name: 'referralSource',
+      QuestionChoices: [
+        { name: 'Friend', id: 'Friend' },
+        { name: 'Facebook', id: 'Facebook' },
+        { name: 'LinkedIn', id: 'LinkedIn' },
+        { name: 'Search Engine', id: 'Search Engine' },
+      ],
+    },
+  ]
+  useEffect(() => {
+    ;(async (): Promise<void> => {
+      try {
+        await getData()
+      } catch (error) {
+        console.log(error)
+      }
+    })()
+  }, [])
+  useEffect(() => {
+    const frameWorks = tracksValue.flatMap((track, frameworks) => track.frameworks)
+    setFrameWorks(frameWorks)
+  }, [tracksValue])
+  const getData = async (): Promise<void> => {
+    const { data, error } = await getHandler({ endpoint: '/submissions' })
+    if (error) return handleError(error)
+    setAllFieldsData(data)
+  }
+
+  const generateBody = ({
+    tracks,
+    employmentStatus,
+    frameworks,
+    referralSource,
+    preferredLanguage,
+  }: IFinishSignUpFrom) => {
+    return {
+      trackId: tracks[0].id,
+      trackName: tracks[0].name,
+      frameworkId: frameworks[0].id,
+      frameworkName: frameworks[0].name,
+      referralSource: referralSource[0].name,
+      employmentStatus: employmentStatus[0].name,
+      preferredLanguage: preferredLanguage[0].name,
+    }
+  }
+  const submitHandler = async (body: IFinishSignUpFrom) => {
+    const requestBody = generateBody(body)
+    const { error } = await postHandler({ endpoint: '/submissions/phaseone', body: requestBody })
+    if (error) return handleError(error)
+  }
+
+  const submit = (e: FormEvent<HTMLFormElement>) => {
```

Comment:
why not calling api in this function ?

Reply:
i think i cant cuz should i  use prevent defualt first